### PR TITLE
Added gaussian1d tests

### PR DESCRIPTION
**Pull request by anchitjain1234**
There is an issue in ``set_masks`` because it does not change the masks, that's why I commented it out.
Also have to add tests for ``fdf`` and ``add`` functions, but I am not sure how they work.

### CODE REVIEW

**Comment by anchitjain1234** [comment]
@gijzelaerr @tammojan Please review.

**Comment by tammojan** [comment]
Looks excellent to me! Now that you've found your first bug, (`set_masks()` not working) could you file it?

**Comment by tammojan** [comment]
To double the test coverage, you could repeat the tests for `functionals.gaussian1d([1, 2, 3], 1)` (where the second argument denotes that this is a complex functional).

**Comment by gijzelaerr** [comment]
This looks very good, great work!

One remark, when writing unit tests you should not use the assert statement but for example the assertTrue method from unittest, so self.assertTrue in this case.

There are all kind of assertions you can make, for example if a certain statement should generate an exception:

for example something liket his:
```
with self.assertRaises(Exception):
    raise_exception()
```

Please change this, commit and then squash all your commits into one commit.

**Comment by anchitjain1234** [comment]
@gijzelaerr Please check now.

File: tests/test_funcionals.py [hunk 1]
```diff
@@ -7,3 +7,31 @@ def test_something(self):
...
+        gauss1d.set_mask(0, False)
+        self.assertTrue(gauss1d.get_masks() == [False, True, True])
+        # gauss1d.set_masks([True, True, True])
+        # self.assertTrue(gauss1d.get_masks() == [True, True, True]
+
+        gauss1d.set_parameters([0, 0, 1])
+        self.assertTrue(gauss1d.f(0) == [0])
+
+    def test_gaussian1d_dtype(self):
+        self.test_gaussian1d(1)
```
Comment (tammojan) [inline tests/test_funcionals.py]: Perhaps add `self.test_gaussian1d(0)` here.

File: tests/test_funcionals.py [hunk 2]
```diff
@@ -7,3 +7,31 @@ def test_something(self):
...
+        gauss1d.set_parameters([3, 4, 5])
+        self.assertTrue(gauss1d.get_parameters() == [3, 4, 5])
+        gauss1d.set_parameter(0, 2)
+        self.assertTrue(gauss1d.get_parameters()[0] == 2)
+
+        self.assertTrue(gauss1d.get_masks() == [True, True, True])
+        gauss1d.set_mask(0, False)
+        self.assertTrue(gauss1d.get_masks() == [False, True, True])
+        # gauss1d.set_masks([True, True, True])
+        # self.assertTrue(gauss1d.get_masks() == [True, True, True]
```
Comment (tammojan) [inline tests/test_funcionals.py]: Could you can add a link (in a comment) to https://github.com/casacore/casacore/pull/327 or something

**Comment by tammojan** [comment]
@gijzelaer Can you merge this?

**Pull request merged**